packages/create-instance/create-slot-vnodes.js

```diff
@@ -4,39 +4,32 @@ import { compileToFunctions } from 'vue-template-compiler'
 
 function createVNodes (
   vm: Component,
-  slotValue: string
+  slotValue: string,
+  name
 ): Array<VNode> {
-  const el = compileToFunctions(`<div>${slotValue}</div>`)
+  const el = compileToFunctions(
+    `<div><template slot=${name}>${slotValue}</template></div>`
+  )
   const _staticRenderFns = vm._renderProxy.$options.staticRenderFns
   const _staticTrees = vm._renderProxy._staticTrees
   vm._renderProxy._staticTrees = []
   vm._renderProxy.$options.staticRenderFns = el.staticRenderFns
   const vnode = el.render.call(vm._renderProxy, vm.$createElement)
   vm._renderProxy.$options.staticRenderFns = _staticRenderFns
   vm._renderProxy._staticTrees = _staticTrees
-  return vnode.children
+  return vnode.children[0]
 }
 
 function createVNodesForSlot (
   vm: Component,
   slotValue: SlotValue,
   name: string,
 ): VNode | Array<VNode> {
-  let vnode
   if (typeof slotValue === 'string') {
-    const vnodes = createVNodes(vm, slotValue)
-    if (vnodes.length > 1) {
-      return vnodes
-    }
-    vnode = vnodes[0]
-  } else {
-    vnode = vm.$createElement(slotValue)
-  }
-  if (vnode.data) {
-    vnode.data.slot = name
-  } else {
-    vnode.data = { slot: name }
+    return createVNodes(vm, slotValue, name)
   }
+  const vnode = vm.$createElement(slotValue)
+  ;(vnode.data || (vnode.data = {})).slot = name
```

Comment:
what does this line do?

Reply:
It sets `vnode.data.slot` to equal the slot name. It's shorthand for:

```js
if(!vnode.data) {
  vnode.data = {}
}
vnode.data.slot = name
```

Reply:
I see! haven't seen this syntax before. I take it the preceeding `;` is required.

Reply:
Yes, that's is one of the drawbacks of not including semi-colons, there are a few other cases where no semi-colons will cause errors.